cpp/src/arrow/util/bit-util.h

```diff
@@ -54,9 +54,12 @@ class BitmapReader {
  public:
   BitmapReader(const uint8_t* bitmap, int64_t start_offset, int64_t length)
       : bitmap_(bitmap), position_(0), length_(length) {
+    current_byte_ = 0;
     byte_offset_ = start_offset / 8;
     bit_offset_ = start_offset % 8;
-    current_byte_ = bitmap[byte_offset_];
+    if (length > 0) {
+      current_byte_ = bitmap[byte_offset_];
+    }
```

Comment:
Can you add a test for this case in bit-util-test.cc? The invalid access will show up in valgrind